guides/source/asset_pipeline.md

```diff
@@ -868,218 +906,350 @@ each asset will have a different file name when it is changed. This way you
+
+## Advanced Asset Management
+
+Over the years, there have been multiple default approaches for handling assets.
+As the web evolved, we began to see more JavaScript-heavy applications.
+
+There is no one-size-fits-all solution for the various JavaScript and CSS
+frameworks and extensions available. However, there are other bundling libraries
+in the Rails ecosystem that should empower you in cases where the default setup
+isn't enough.
+
+### `jsbundling-rails`
+
+[`jsbundling-rails`](https://github.com/rails/jsbundling-rails) is a gem that
+integrates modern JavaScript bundlers into your Rails application. It allows you
+to manage and bundle JavaScript assets with tools like [Bun](https://bun.sh),
+[esbuild](https://esbuild.github.io/), [rollup.js](https://rollupjs.org/), or
+[Webpack](https://webpack.js.org/), offering a runtime-dependent approach for
+developers seeking flexibility and performance.
+
+#### How `jsbundling-rails` Works
+
+1. After installation, it sets up your Rails app to use your chosen JavaScript
+   bundler.
+2. It creates a `build` script in your `package.json` file to compile your
+   JavaScript assets.
+3. During development, the `build:watch` script ensures live updates to your
+   assets as you make changes.
+4. In production, the gem ensures that JavaScript is built and included during
+   the precompilation step, reducing manual intervention. It hooks into Rails'
+   `assets:precompile` task to build JavaScript for all entry points during
+   deployment. This integration ensures that your JavaScript is production-ready
+   with minimal configuration.
+
+The gem automatically handles entry-point discovery - identifying the primary
+JavaScript files to bundle by following Rails conventions, typically looking in
+directories like `app/javascript/` and configuration. By adhering to Rails
+conventions, `jsbundling-rails` simplifies the process of integrating complex
+JavaScript workflows into Rails projects.
+
+#### When Should You Use It?
+
+`jsbundling-rails` is ideal for Rails applications that:
+
+- Require modern JavaScript features like ES6+, TypeScript, or JSX.
+- Need to leverage bundler-specific optimizations like tree-shaking, code
+  splitting, or minification.
+- Use `Propshaft` for asset management and need a reliable way to integrate
+  precompiled JavaScript with the broader Rails asset pipeline.
+- Utilize libraries or frameworks that depend on a build step. For example,
+  projects requiring transpilation—such as those using
+  [Babel](https://babeljs.io/), [TypeScript](https://www.typescriptlang.org/),
+  or React JSX—benefit greatly from `jsbundling-rails`. These tools rely on a
+  build step, which the gem seamlessly supports.
+
+By offering deep integration with Rails tools like `Propshaft` and simplifying
+JavaScript workflows, `jsbundling-rails` empowers you to build rich, dynamic
+front-ends while staying productive and adhering to Rails conventions.
```

Comment:
This paragraph reads more like a marketing site for `jsbundling-rails` 😉 
Maybe we should make it a bit more plain, for example:
```suggestion
By integrating with Rails tools like `Propshaft` and simplifying
JavaScript workflows, `jsbundling-rails` allows you to build rich, dynamic
front-ends while staying productive and adhering to Rails conventions.
```